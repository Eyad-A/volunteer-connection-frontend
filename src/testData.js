import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
    username: "demouser",
    first_name: "demouser",
    last_name: "demouser",
    email: "demouser@demouser.com",
    skill: "Developer"
};

const UserProvider = ({
    children, currentUser = demoUser, hasConnectedToCompany = () => false 
}) => (
    <UserContext.Provider value={{ currentUser, hasConnectedToCompany }}>
        {children}
    </UserContext.Provider>
);

export { UserProvider};