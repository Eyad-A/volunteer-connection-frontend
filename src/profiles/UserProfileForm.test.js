import React from "react";
import { render } from "@testing-library/react";
import UserProfile from "./UserProfileForm";
import { UserProvider } from "../testData";

it("matches snapshot", function () {
  const { asFragment } = render(
      <UserProvider>
        <UserProfile />
      </UserProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});