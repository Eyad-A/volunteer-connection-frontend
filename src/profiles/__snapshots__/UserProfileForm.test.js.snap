// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container my-5"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-2"
      />
      <div
        class="col-lg-10"
      >
        <h1>
          User Profile Form
        </h1>
        <div
          class="my-3"
        >
          <form>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                Username
              </label>
              <p>
                demouser
              </p>
            </div>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                First name
              </label>
              <input
                class="form-control"
                name="firstName"
                value=""
              />
            </div>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                Last name
              </label>
              <input
                class="form-control"
                name="lastName"
                value=""
              />
            </div>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                Skill
              </label>
              <input
                class="form-control"
                name="skill"
                value="Developer"
              />
            </div>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                Email
              </label>
              <input
                class="form-control"
                name="email"
                value="demouser@demouser.com"
              />
            </div>
            <div
              class="col-lg-12 my-2"
            >
              <label>
                Confirm password to make changes
              </label>
              <input
                class="form-control"
                name="password"
                type="password"
                value=""
              />
            </div>
            <button
              class="btn btn-lg btn-primary my-3"
            >
              Save
            </button>
          </form>
        </div>
      </div>
      <div
        class="col-lg-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
